cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryRedirect.ts

```diff
@@ -1,20 +1,58 @@
-import { CloudFrontRequestHandler } from 'aws-lambda';
+import {CloudFrontRequestEvent, CloudFrontRequestHandler, CloudFrontResultResponse, Context} from 'aws-lambda';
 import { CloudFrontRequest } from 'aws-lambda/common/cloudfront';
+import axios from 'axios';
+import assert from "assert";
+import { CatalogueResultsList } from './catalogue'
 
-export const request: CloudFrontRequestHandler = (event, context, callback) => {
+async function apiFetch(path: string) {
+  return axios.get(
+    'https://api.wellcomecollection.org/catalogue/v2' + path
+  )
+}
+
+// TODO: Account for multiple results by looking for matching sierra entry
+// TODO: Account for situations where results overflow page
+async function getWork(bNumber: string) {
+  const apiPath = `/works?identifiers=${bNumber}`
+  const results = await apiFetch(apiPath)
+
+  const resultList = results.data as CatalogueResultsList
+  assert(resultList.results.length >= 1, "No matching Catalogue API results found")
```

Comment:
Is an assertion the best behaviour here? Wouldn't it be better to return a 404 to the user?

Reply:
Assert exits with a non-0 code, which in these lambdas, we should never do as that will send the 500 on the cache layer.

Reply:
Obvs.

Reply:
[Assuming that this is saying to deal with this more gracefully](https://github.com/wellcomecollection/platform-infrastructure/pull/116/files#diff-cad288977e3edeafbbc2ad132d4fd147a6762ba2623c35f9fc6379e3d638d2d3R38)

I would also say it's odd practise in terms of it not really being errors, but rather known states of paths we would redirect.

The more common approach would be to return either undefined or a reason if that's needed later for logging, but you could just do the logging here.